scipy/signal/lti_conversion.py

```diff
@@ -146,6 +146,12 @@ def _restore(M, shape):
         return M
 
 
+def _inexact(m):
+    if not np.issubdtype(m.dtype, np.inexact):
+        m = m.astype(np.float_)
```

Comment:
`np.float_` was removed in the NumPy `2.0` release; could perhaps replace with `np.float64` if we still want this approach. Also worth noting that this module is now "private" at `_lti_conversion.py` so when I "rebased" locally I basically manually moved the changes over.